File: unidade01/conceitos/estrutural/paciente.js
export class Paciente {
  #nome;
  #idade;
  constructor(nome, idade) {
    this.#nome = nome;
    this.#idade = idade;
  }

  getNome() {
    return this.#nome;
  }

  getIdade() {
    return this.#idade;
  }

  setNome(nome) {
    this.#nome = nome;
    return this.nome;
  }

  setIdade(idade) {
    this.#idade = idade;
    return this.#idade;
  }

  solicitar_consulta(medico) {
    if (!medico) {
      throw Error("Médico não localizado.");
    }
    console.log(
      `O paciente: ${this.#nome} solicitou uma consulta com o médico: ${medico.getNome()}`,
    );
  }
}

File: estudo-aoo/src/estrutural/Paciente.java
package estrutural;
public class Paciente {
    // Atributos
    private String nome;
    private int idade;

    // Construtor
    public Paciente(String nome, int idade){
        this.nome = nome;
        this.idade = idade;
    }

    // Getters e Setters
    public String getNome() {
        return nome;
    }

    public int getIdade(){
        return      idade;
    }

    public void setNome(String nome){
        this.nome = nome;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public void solicitarConsulta(Medico medico){
        System.out.println("Paciente: " +
                this.getNome() + " solicitou uma consulta com " +
                medico.getNome());
        medico.realizarConsulta(this);
    }
}
